knack/cli.py

```diff
@@ -219,6 +219,7 @@ def invoke(self, args, initial_invocation_data=None, out_file=None):
                 if cmd_result and cmd_result.result is not None:
                     formatter = self.output.get_formatter(output_type)
                     self.output.out(cmd_result, formatter=formatter, out_file=out_file)
+            self.raise_event(EVENT_CLI_SUCCESSFUL_EXECUTE)
```

Comment:
two concerns:
1. could the new event bring benefit to CLI if exit code = 0 could represent successful execution?
2.  should SUCCESSFUL_EXECUTE be raised after cmd_result is returned? It seems that in current code output format failure would block the event.

Reply:
> could the new event bring benefit to CLI if exit code = 0 could represent successful execution?

I couldn't really get your question. Could you elaborate?

> should SUCCESSFUL_EXECUTE be raised after cmd_result is returned?

No. If `self.output.out` failed, this is considered a failure and `SUCCESSFUL_EXECUTE` shouldn't be raised by design.